test/CadlRanchProjects.Tests/OAuth2TestHelper.cs

```diff
@@ -50,13 +55,55 @@ public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePol
             protected new async ValueTask ProcessNextAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
             {
                 await _transport.ProcessAsync(message).ConfigureAwait(false);
+                await ResponseBodyPolicyAsync(message, pipeline).ConfigureAwait(false);
 
                 var response = message.Response;
                 Type responseType = response.GetType();
                 PropertyInfo propInfo = responseType.GetProperty("IsError", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
                 propInfo.SetValue(response, message.ResponseClassifier.IsErrorResponse(message));
             }
 
+            private async ValueTask ResponseBodyPolicyAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
```

Comment:
consider to put this together with `CopyToAsync` and add comments where they are from.